forc-plugins/forc-fmt/src/main.rs

```diff
@@ -104,49 +260,13 @@ fn format_pkg_at_dir(app: &App, dir: &Path, formatter: &mut Formatter) -> Result
             let mut contains_edits = false;
 
             for file in files {
-                if let Ok(file_content) = fs::read_to_string(&file) {
-                    let file_content: Arc<str> = Arc::from(file_content);
-                    let build_config = BuildConfig::root_from_file_name_and_manifest_path(
-                        file.clone(),
-                        manifest_path.clone(),
-                        BuildTarget::default(),
-                    );
-                    match Formatter::format(formatter, file_content.clone(), Some(&build_config)) {
-                        Ok(formatted_content) => {
-                            if app.check {
-                                if *file_content != formatted_content {
-                                    contains_edits = true;
-                                    info!("\n{:?}\n", file);
-                                    display_file_diff(&file_content, &formatted_content)?;
-                                }
-                            } else {
-                                format_file(&file, &formatted_content)?;
-                            }
-                        }
-                        Err(err) => {
-                            // there could still be Sway files that are not part of the build
-                            error!("\nThis file: {:?} is not part of the build", file);
-                            error!("{}\n", err);
-                        }
-                    }
-                }
+                if let Ok(edited) = format_file(app, file, Some(manifest_file.clone()), formatter) {
+                    contains_edits = edited;
+                };
             }
             // format manifest using taplo formatter
-            if let Ok(file_content) = fs::read_to_string(&manifest_file) {
-                let taplo_alphabetize = taplo_fmt::Options {
-                    reorder_keys: true,
-                    ..Default::default()
-                };
-                let formatted_content = taplo_fmt::format(&file_content, taplo_alphabetize);
-                if !app.check {
-                    format_file(&manifest_file, &formatted_content)?;
-                } else if formatted_content != file_content {
-                    contains_edits = true;
-                    error!("\nManifest Forc.toml improperly formatted");
-                    display_file_diff(&file_content, &formatted_content)?;
-                } else {
-                    info!("\nManifest Forc.toml properly formatted")
-                }
+            if let Ok(edited) = format_manifest(app, manifest_file) {
+                contains_edits = edited;
```

Comment:
If the `format_file` returns true and `format_manifest` returns false, we still want `contains_edits` to be true, right?

```suggestion
            if let Ok(edited) = format_manifest(app, manifest_file) {
                contains_edits = contains_edits || edited;
```

Reply:
good catch!